service/raft/group0_state_machine.cc

```diff
@@ -53,26 +54,156 @@ namespace service {
 
 static logging::logger slogger("group0_raft_sm");
 
-group0_state_machine::group0_state_machine(raft_group0_client& client, migration_manager& mm, storage_proxy& sp, storage_service& ss, raft_address_map& address_map, gms::feature_service& feat, bool topology_change_enabled)
-        : _client(client), _mm(mm), _sp(sp), _ss(ss), _address_map(address_map), _topology_change_enabled(topology_change_enabled)
-        , _topology_on_raft_support_listener(feat.supports_consistent_topology_changes.when_enabled([this] () noexcept {
-            // Using features to decide whether to start fetching topology snapshots
-            // or not is technically not correct because we also use features to guard
-            // whether upgrade can be started, and upgrade starts by writing
-            // to the system.topology table (namely, to the `upgrade_state` column).
-            // If some node at that point didn't mark the feature as enabled
-            // locally, there is a risk that it might try to pull a snapshot
-            // and will decide not to use `raft_pull_snapshot` verb.
-            //
-            // The above issue is mitigated by requiring administrators to
-            // wait until the SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES feature
-            // is enabled on all nodes.
-            //
-            // The biggest value of using a cluster feature here is so that
-            // the node won't try to fetch a topology snapshot if the other
-            // node doesn't support it yet.
-            _topology_change_enabled = true;
-        })) {
+class group0_state_id_handler : public gms::i_endpoint_state_change_subscriber, public enable_shared_from_this<group0_state_id_handler> {
+
+    std::reference_wrapper<db::system_keyspace> _sys_ks;
+    std::reference_wrapper<replica::database> _local_db;
+    std::reference_wrapper<gms::gossiper> _gossiper;
+
+    bool _stopped = false;
+
+    std::unordered_map<gms::inet_address, utils::UUID> _state_ids;
```

Comment:
I'm wondering, why don't we use boost::unordered_flat_map, which is almost always [faster](https://www.boost.org/doc/libs/1_85_0/libs/unordered/doc/html/unordered.html#benchmarks) than std unordered map?

Reply:
This is control plane path, not performance critical. It wouldn't make a difference. And more C++ developers are familiar with `std` data structures than `boost` ones.